File: backend/app/api/v1/schemas/project.py
# app/api/v1/schemas/project.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
import uuid


class ProjectCreate(BaseModel):
    """Données nécessaires pour créer un projet."""
    name: str
    description: Optional[str] = None


class ProjectUpdate(BaseModel):
    """Données modifiables d'un projet."""
    name: Optional[str] = None
    description: Optional[str] = None


class ProjectResponse(BaseModel):
    """Ce qui est renvoyé au frontend."""
    id: uuid.UUID
    name: str
    description: Optional[str]
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True  # Permet la conversion depuis l'ORM


class ProjectListResponse(BaseModel):
    """Liste de projets avec compteur."""
    projects: list[ProjectResponse]
    total: int

File: frontend/src/types/project.ts
/**
 * Types et interfaces relatifs aux projets de transcription.
 */

export interface Project {
  id: string;
  name: string;
  description: string | null;
  owner_id: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreateInput {
  name: string;
  description?: string;
}

export interface ProjectUpdateInput {
  name?: string;
  description?: string;
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
}

export type SortOption = "date_desc" | "date_asc" | "name_asc" | "name_desc";
